### ISSUE TITLE
How to reproduce the result for the simulated data? What is the reasonable value of loss?

### ISSUE THREAD

**Issue by redLinmumu** [yutongo/TIGON#5]
I try to reproduce the result on the  simulated data, but the losses are too much(more than 1 million).  The args I use are as follows(I found them  in the supplementary files):
dataset -> Simulation 
n_hiddens -> 3 
niters  -> 3000,
sigma  -> 1.0 (sigma is set in the code directly. I also try sigma=0.3, and it doesn't work, too.),
hidden_dim  -> 16,
num_samples -> 1000,
timepoints -> [0 ,10,  20,  30, 40] (consistent with Fig 2.) 
lr  -> 0.05.
How to reproduce the result for the simulated data? What is the reasonable value of loss?
Is there anything wrong with the args? Or am I losing some important details?
I used above hyperparameters, and I run the training.ipynb file, but the loss value printed in the log  is  far too large(more than 1 million), which seems to be unreasonable apparently. I did not load the existed model. 
Now I am so confused. 

Here is what the loss looks like:
Iter: 2582, loss: 118904508286809938251329992392704.0000.

I really appreciate it if you can help me with reproducing the result. Thanks a lot!

**Comment by YuchiQiu** [comment]
Did you see the loss blow up? The simulated data is kind of stiff, which is difficult to train. Could you try multiple runs and see how many of them have this issue? Also try to make the maximal epochs smaller and see if it converges.

**Comment by redLinmumu** [comment]
How many runs have you tried? Can you give me your settings of max epochs and any other settings? I will try with your settings directly.

---- Replied Message ----
| From | ***@***.***> |
| Date | 03/23/2025 11:49 |
| To | yutongo/TIGON ***@***.***> |
| Cc | Fengjiao_Gong ***@***.***>,
Author ***@***.***> |
| Subject | Re: [yutongo/TIGON] How to reproduce the result for the simulated data? What is the reasonable value of loss? (Issue #5) |

Did you see the loss blow up? The simulated data is kind of stiff, which is difficult to train. Could you try multiple runs and see how many of them have this issue? Also try to make the maximal epochs smaller and see if it converges.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

YuchiQiu left a comment (yutongo/TIGON#5)

Did you see the loss blow up? The simulated data is kind of stiff, which is difficult to train. Could you try multiple runs and see how many of them have this issue? Also try to make the maximal epochs smaller and see if it converges.
